lib/pbench/server/api/resources/query_apis/datasets/datasets_contents.py

```diff
@@ -65,14 +67,20 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
 
         # Retrieve the ES indices that belong to this run_id from the metadata
         # table
+        try:
+            indices = self.get_index(dataset, "run-toc")
+        except InvalidAPIRequest:
+            raise InvalidAPIRequest(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                "Indices from metadata table not found",
+            )
 
-        indices = self.get_index(dataset, "run-toc")
         if not indices:
             self.logger.debug(
                 f"Found no indices matching the prefix run-toc"
                 f" for a dataset {dataset!r}"
             )
-            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+            raise InvalidAPIRequest(HTTPStatus.NOT_FOUND, f"No matching indices found")
```

Comment:
Again, not really your problem since it already existed; but since `get_index` already raises `InvalidAPIRequest` if the metadata value is "falsey" there's no need for this block at all. (In fact, it should be impossible to get here, and I wonder why our tools aren't flagging that.)

Reply:
So we can just remove the entire block from line 78 to 83?

Sounds good to me.

Reply:
If we don't remove this block, then we Flake8 would like us to fix the f-string which contains no format parameters....